src/remote_adapter/remote_adapter_server/main.cc

```diff
@@ -0,0 +1,315 @@
+
+                  const int res = UtilsManager::FileUpdate(
+                                                    file_path,
+                                                    file_name,
+                                                    file_content);
+                  return res;
+             });
+
+    srv.bind(constants::file_exists,
+             [](std::string file_path,std::string file_name){
+               const int res = UtilsManager::FileExists(
+                                                    file_path,
+                                                    file_name);
+               return res;
+             });
+
+    srv.bind(constants::file_delete,
+             [](std::string file_path,std::string file_name){
+               const int res = UtilsManager::FileDelete(
+                                                    file_path,
+                                                    file_name);
+               return res;
+             });
+
+    srv.bind(constants::file_content,
+             [](std::string file_path,
+                std::string file_name,
+                size_t offset,
+                size_t max_size_content){
+
+                  std::string file_content =
+                           UtilsManager::GetFileContent(
+                                          file_path,
+                                          file_name,
+                                          offset,
+                                          max_size_content
+                                          );
+
+                  return std::make_pair(
+                                  file_content,
+                                  offset
+                                  );
+             });
+
+    srv.bind(constants::folder_exists,
+             [](std::string folder_path){
+               const int res = UtilsManager::FolderExists(
+                                                    folder_path);
+               return res;
+             });
+
+
+
+
+
+    srv.bind(constants::folder_delete,
+             [](std::string folder_path){
+               const int res = UtilsManager::FolderDelete(
+                                                    folder_path);
+               return res ?
+                constants::error_codes::FAILED
+                :
+                constants::error_codes::SUCCESS;
+             });
+
+    srv.bind(constants::folder_create,
+             [](std::string folder_path){
+               const int res =
+                  UtilsManager::FolderCreate(folder_path);
+               return res;
+             });
+
+    srv.bind(constants::command_execute,
+             [](std::string bash_command){
+
+                auto receive_result =
+                            UtilsManager::ExecuteCommand(bash_command);
+
+                return receive_result;
+
+             });
+
+
+    srv.suppress_exceptions(true);
+    // Run the server loop with 1 worker threads.
+    srv.async_run();
```

Comment:
@SNiukalov Why does server use only one worker thread? As I understand, it is needed to use 2 worker threads (one for receive rpc and one for other "fast" rpcs) isn't it?

Reply:
@aderiabin Server ia working in asynchronous mode now.